src/components/AddCollaboratorForm.tsx

```diff
@@ -0,0 +1,127 @@
+import { getCollaborators, addCollaborator, getAdmin } from "@/actions/list";
+
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuLabel,
+  DropdownMenuSeparator,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+
+import { useEffect, useState } from "react";
+
+interface Collaborator {
+  Collaborator: string[];
+}
+
+interface AddCollaboratorProps {
+  listId: string;
+  onSubmitSuccess: () => void;
+}
+
+export const AddCollaborator = ({
+  listId,
+  onSubmitSuccess,
+}: AddCollaboratorProps) => {
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [collaborator, setCollaborator] = useState<Collaborator | null>(null);
+  const [admin, setAdmin] = useState<string | null>(null);
+  const [isClicked, setIsClicked] = useState(false);
+
+  useEffect(() => {
+    const fetchAdmin = async () => {
+      try {
+        const data = await getAdmin(listId);
```

Comment:
Would be better if this was queried in a server component then passed here as a prop. Can do it next time